### PR TITLE
Sema: Only add `@_spiOnly` to import fix-its with a public access level

### PR DESCRIPTION
**Pull request by tshortli**
It doesn't make sense to add the `@_spiOnly` attribute to an `internal` import.

Part of rdar://126637855.

### CODE REVIEW

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Comment by tshortli** [comment]
@swift-ci please smoke test Linux

File: test/NameLookup/members_transitive_multifile_access_level.swift [hunk 1]
```diff
@@ -8,6 +8,9 @@
 // RUN: %target-swift-frontend -emit-module -o %t %t/MixedUses.swift
 // RUN: %target-swift-frontend -emit-module -o %t %t/InternalUsesOnlyTransitivelyImported.swift
 // RUN: %target-swift-frontend -emit-module -o %t %t/Exports.swift -I %t
+// RUN: %target-swift-frontend -emit-module -o %t %t/InternalUsesOnlySPIOnly.swift -I %t
+// RUN: %target-swift-frontend -emit-module -o %t %t/InternalUsesOnlyDefaultedImportSPIOnly.swift -I %t
+// RUN: %target-swift-frontend -emit-module -o %t %t/PublicUsesOnlySPIOnly.swift -I %t
```
Comment (xymus) [inline test/NameLookup/members_transitive_multifile_access_level.swift:13]: Do we have tests for the `InternalImportsByDefault` use case related to this feature?

**Review by xymus — APPROVED** [review]

File: test/NameLookup/members_transitive_multifile_access_level.swift (hunk 1 above)
Comment (tshortli) [reply test/NameLookup/members_transitive_multifile_access_level.swift:13]: Not yet, but I ought to cover that case.

Comment (tshortli) [reply test/NameLookup/members_transitive_multifile_access_level.swift:13]: Added test coverage in the latest version.

**Pull request merged**